include/dr/details/segments_tools.hpp

```diff
@@ -4,52 +4,93 @@
 
 #pragma once
 
+#include <dr/concepts/concepts.hpp>
+#include <dr/details/device_subrange.hpp>
+#include <dr/details/enumerate.hpp>
+#include <dr/details/ranges_shim.hpp>
+#include <dr/details/view_detectors.hpp>
+
 namespace lib {
 
 namespace internal {
 
-// return number of full segments and remainder to cover n elements
+// count the number of segments necessary to cover n elements,
+// returning the index of the last segment and its remainder
 template <typename R>
-void n_segs_remainder(R &&segments, std::size_t n, auto &n_segs,
+void n_segs_remainder(R &&segments, std::size_t n, auto &last_seg,
```

Comment:
I made a few changes to segments tools to avoid leaving an empty segment at the end each time as well as to use the new distributed iterator design.